### DISCUSSION TITLE
Get `Goto definition for Java classes will not work` error.

### DISCUSSION THREAD

**Discussion by HaoYang670** [scalameta/metals#5164] [Q&A] [answered]
Hi.

I get `Goto definition for Java classes will not work` error when using scala metals on vscode:
<img width="661" alt="image" src="https://user-images.githubusercontent.com/59198230/233536657-583246b0-0788-4250-9bef-aa09733a166c.png">

I have setup the JAVA_HOME on  my computer:
```
$ echo "$JAVA_HOME"
/usr/lib/jvm/java-8-openjdk-amd64
```
when go into the build target, it shows me this:
<img width="1523" alt="image" src="https://user-images.githubusercontent.com/59198230/233536925-62b7c4e3-ea1e-43ea-a57f-f0cdb7de68f6.png">

How to fix it? Need your help.

**Comment by HaoYang670** [comment]
I find this error from the log
```
2023.04.21 15:50:53 ERROR Maven command failed: /usr/bin/apache-maven-3.8.5/bin/mvn generate-sources ch.epfl.scala:bloop-maven-plugin:2.0.0:bloopInstall -DdownloadSources=true
```

**Comment by tgodzik** [comment, marked as answer]
JDK source are not installed by default with Java 8 I think. You need to install them manually. If you are using ubuntu you can use `sudo apt-get install openjdk-8-source` I think.

**Reply by HaoYang670** [reply]
Thank you @tgodzik. You advise partially fix my problem. Thank you!

But I got a new error from the log:
```
org.apache.thrift:libfb303:jar:sources:0.9.3 was not found in https://maven-central-asia.storage-download.googleapis.com/maven2/ during a previous attempt.
```

**Reply by tgodzik** [reply]
That source jar might actually not exist, but this might only be an issue if you want to go to sources for that jar.

**Reply by HaoYang670** [reply]
Is there any way to provide an alternative maven central? I set the config `Coursier Mirror`, but it doesn't work

**Reply by tgodzik** [reply]
We use whatever maven uses in this instance. Metals only invokes maven via shell, so everything should work ok.

**Reply by HaoYang670** [reply]
Thank you @tgodzik, it was actually a network problem, and Metals works now!

**Reply by markusa380** [reply]
How do I get the sources if I am on MacOS and installed OpenJDK 17 through brew?

**Reply by tgodzik** [reply]
They should be installed along with JDK 17, I think the issues were usually with JDK 8. Are you having warnings in the Metals Doctor about that?

**Reply by markusa380** [reply]
@tgodzik I am getting warnings in the Metals doctor, but I am using JDK 17:

```
java -version
openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment Homebrew (build 17.0.6+0)
OpenJDK 64-Bit Server VM Homebrew (build 17.0.6+0, mixed mode, sharing)
```

and the `Java Home` extension setting is empty.

Output of the doctor:

```
Metals Java: 17.0.6 from Homebrew located at /opt/homebrew/Cellar/openjdk@17/17.0.6/libexec/openjdk.jdk/Contents/Home
```

....

Build target | Type | Compilation status | Diagnostics | Interactive | Semanticdb | Debugging | Java support | Recommendation
-- | -- | -- | -- | -- | -- | -- | -- | --
root | Scala 2.13.8 | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | Goto definition for Java classes will not work, please install jdk sources in java home. Searched: /opt/homebrew/opt/src.zip, /opt/homebrew/opt/lib/src.zip, /opt/homebrew/opt/openjdk@17/src.zip, /opt/homebrew/opt/openjdk@17/lib/src.zip
root-test | Scala 2.13.8 | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | Goto definition for Java classes will not work, please install jdk sources in java home. Searched: /opt/homebrew/opt/src.zip, /opt/homebrew/opt/lib/src.zip, /opt/homebrew/opt/openjdk@17/src.zip, /opt/homebrew/opt/openjdk@17/lib/src.zip
grpc-deps | Scala 2.13.8 | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | Goto definition for Java classes will not work, please install jdk sources in java home. Searched: /opt/homebrew/opt/src.zip, /opt/homebrew/opt/lib/src.zip, /opt/homebrew/opt/openjdk@17/src.zip, /opt/homebrew/opt/openjdk@17/lib/src.zip
grpc-deps-test | Scala 2.13.8 | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | Goto definition for Java classes will not work, please install jdk sources in java home. Searched: /opt/homebrew/opt/src.zip, /opt/homebrew/opt/lib/src.zip, /opt/homebrew/opt/openjdk@17/src.zip, /opt/homebrew/opt/openjdk@17/lib/src.zip
eng-xxxxxxxxx-build | sbt 1.6.1 | ✅ | ⚠️ | ✅ | ✅ | ❌ | ✅ | Goto definition for Java classes will not work, please install jdk sources in java home. Searched: /opt/homebrew/opt/src.zip, /opt/homebrew/opt/lib/src.zip, /opt/homebrew/opt/openjdk@17/src.zip, /opt/homebrew/opt/openjdk@17/lib/src.zip
eng-xxxxxxxxxxxx-build-build | sbt 1.6.1 | ✅ | ⚠️ | ✅ | ✅ | ❌ | ✅ | Goto definition for Java classes will not work, please install jdk sources in java home. Searched: /opt/homebrew/opt/src.zip, /opt/homebrew/opt/lib/src.zip, /opt/homebrew/opt/openjdk@17/src.zip, /opt/homebrew/opt/openjdk@17/lib/src.zip

**Reply by tgodzik** [reply]
Is the sources jar present there? Somewhere in the home?

**Reply by markusa380** [reply]
I don't think so, @tgodzik . I don't think they are included, are they?

**Reply by tgodzik** [reply]
They should, if they aren't there is no sensible way to download them :/

**Reply by khanhbq2k** [reply]
I'm using jdk 17 also and the src.zip file is not included in the installation, so i downloaded, zipped and move it manually to the lib folder
Source iam using for arch: https://archlinux.org/packages/extra/x86_64/openjdk17-src/
